strapi/strapi
# fix(i18n): Do not copy data if it is not visible

### What does it do?

It makes sure that no invisible fields are copied over from one locale to another.

### Why is it needed?

When a user uses the "fill in from other locale" function, they expect the fields they can see to be copied over. However invisible fields may be used by a plugin or internally by the developer's code. This can then cause an issue where data is copied which should not have been copied.

One of the problems is that invisible fields are not sent to the client, which means that we cannot check if the field is a relation or a password, thus it will be copied over.

### How to test it?

1. Create a content type with some fields
2. Create an entry filling out all the fields
3. In your editor change the `schema.json` file and set visible to false on one of the fields. 
4. Go back to the content manager and use the "fill in from other locale" function
5. See in the database that the data of that field was not copied over.
